gui/notes.lua

```diff
@@ -0,0 +1,381 @@
+-- Map notes
+--@ module = true
+
+local gui = require 'gui'
+local widgets = require 'gui.widgets'
+local guidm = require('gui.dwarfmode')
+local script = require 'gui.script'
+local overlay = require 'plugins.overlay'
+local text_editor = reqscript('internal/journal/text_editor')
+local note_manager = reqscript('internal/notes/note_manager')
+
+local map_points = df.global.plotinfo.waypoints.points
+
+local NOTE_LIST_RESIZE_MIN = {w=26}
+local RESIZE_MIN = {w=65, h=30}
+local NOTE_SEARCH_BATCH_SIZE = 25
+local OVERLAY_NAME = 'notes.map_notes'
+
+local green_pin = dfhack.textures.loadTileset(
+    'hack/data/art/note_green_pin_map.png',
+    32,
+    32,
+    true
+)
+
+NotesSearchField = defclass(NotesSearchField, text_editor.TextEditor)
+NotesSearchField.ATTRS {}
+
+function NotesSearchField:onInput(keys)
+    -- allow cursor up/down to be used to navigate the notes list
+    if keys.KEYBOARD_CURSOR_UP or keys.KEYBOARD_CURSOR_DOWN then
+        return false
+    end
+
+    return NotesSearchField.super.onInput(self, keys)
+end
+
+NotesWindow = defclass(NotesWindow, widgets.Window)
+NotesWindow.ATTRS {
+    frame_title='DF Notes',
+    resizable=true,
+    resize_min=RESIZE_MIN,
+    frame_inset={l=0,r=0,t=0,b=0},
+    on_note_add=DEFAULT_NIL
+}
+
+function NotesWindow:init()
+    self.selected_note = nil
+    self.note_manager = nil
+    self.curr_search_phrase = nil
+
+    self:addviews{
+        widgets.Panel{
+            view_id='note_list_panel',
+            frame={l=0, w=NOTE_LIST_RESIZE_MIN.w, t=0, b=1},
+            visible=true,
+            frame_inset={l=1,t=1,b=1,r=1},
+            autoarrange_subviews=true,
+            subviews={
+                widgets.HotkeyLabel {
+                    key='CUSTOM_ALT_S',
+                    label='Search',
+                    frame={l=0},
+                    auto_width=true,
+                    on_activate=function() self.subviews.search:setFocus(true) end,
+                },
+                NotesSearchField{
+                    view_id='search',
+                    frame={l=0,h=3},
+                    frame_style=gui.FRAME_INTERIOR,
+                    one_line_mode=true,
+                    on_text_change=self:callback('loadFilteredNotes'),
+                    on_submit=function()
+                        self.subviews.note_list:submit()
+                    end
+                },
+                widgets.List{
+                    view_id='note_list',
+                    frame={l=0,b=2},
+                    frame_inset={t=1},
+                    row_height=1,
+                    on_submit=function (ind, note)
+                        self:loadNote(note)
+                        dfhack.gui.pauseRecenter(note.point.pos)
```

Comment:
the flashlight do not work if we render something custom on the map - and we do, the note marker.
any idea how it can be fixed (if it should be fixed)?